Description: Fix undefined loop behaviour.
Author: Petter Reinholdtsen <pere@debian.org>
Bug-Debian: http://bugs.debian.org/753177
Last-Update: 2014-10-01

--- lmms-1.0.3.orig/plugins/LadspaEffect/swh/bode_shifter_cv_1432.c
+++ lmms-1.0.3/plugins/LadspaEffect/swh/bode_shifter_cv_1432.c
@@ -350,7 +350,7 @@ static void runAddingBodeShifterCV(LADSP
 	  /* Perform the Hilbert FIR convolution
 	   * (probably FFT would be faster) */
 	  hilb = 0.0f;
-	  for (i = 0; i <= NZEROS/2; i++) {
+	  for (i = 0; i < NZEROS/2; i++) {
 	    hilb += (xcoeffs[i] * delay[(dptr - i*2) & (D_SIZE - 1)]);
 	  }
 
--- lmms-1.0.3.orig/plugins/LadspaEffect/swh/surround_encoder_1401.c
+++ lmms-1.0.3/plugins/LadspaEffect/swh/surround_encoder_1401.c
@@ -228,7 +228,7 @@ static void runSurroundEncoder(LADSPA_Ha
 	for (pos = 0; pos < sample_count; pos++) {
 	  delay[dptr] = s[pos];
 	  hilb = 0.0f;
-	  for (i = 0; i <= NZEROS/2; i++) {
+	  for (i = 0; i < NZEROS/2; i++) {
 	    hilb += (xcoeffs[i] * delay[(dptr - i*2) & (D_SIZE - 1)]);
 	  }
 	  dptr = (dptr + 1) & (D_SIZE - 1);
@@ -296,7 +296,7 @@ static void runAddingSurroundEncoder(LAD
 	for (pos = 0; pos < sample_count; pos++) {
 	  delay[dptr] = s[pos];
 	  hilb = 0.0f;
-	  for (i = 0; i <= NZEROS/2; i++) {
+	  for (i = 0; i < NZEROS/2; i++) {
 	    hilb += (xcoeffs[i] * delay[(dptr - i*2) & (D_SIZE - 1)]);
 	  }
 	  dptr = (dptr + 1) & (D_SIZE - 1);
